hare/eligibility/oracle.go

```diff
@@ -58,37 +91,57 @@ type Oracle struct {
 	synced bool
 
 	beacons        system.BeaconGetter
-	cdb            *datastore.CachedDB
+	atxsdata       *atxsdata.Data
+	db             sql.Executor
 	vrfVerifier    vrfVerifier
 	layersPerEpoch uint32
-	cfg            config.Config
+	cfg            Config
 	log.Log
 }
 
+type Opt func(*Oracle)
+
+func WithConfig(config Config) Opt {
+	return func(o *Oracle) {
+		o.cfg = config
+	}
+}
+
+func WithLogger(logger log.Log) Opt {
+	return func(o *Oracle) {
+		o.Log = logger
+	}
+}
+
 // New returns a new eligibility oracle instance.
 func New(
 	beacons system.BeaconGetter,
-	db *datastore.CachedDB,
+	db sql.Executor,
+	atxsdata *atxsdata.Data,
 	vrfVerifier vrfVerifier,
 	layersPerEpoch uint32,
-	cfg config.Config,
-	logger log.Log,
+	opts ...Opt,
 ) *Oracle {
-	ac, err := lru.New[types.EpochID, *cachedActiveSet](activesCacheSize)
+	activesCache, err := lru.New[types.EpochID, *cachedActiveSet](activesCacheSize)
 	if err != nil {
-		logger.With().Fatal("failed to create lru cache for active set", log.Err(err))
+		panic("failed to create lru cache for active set" + err.Error())
```

Comment:
i don't think that it is worth. it can fail only if size is negative or zero, that would a programmer error.
ofcourse it can change, but in my opinion panic is appropriate

```
if size <= 0 {
		return nil, errors.New("must provide a positive size")
	}
```